### PR TITLE
Create activitystream list view for clients

### PR DESCRIPTION
**Pull request by Mayaank12**
Task #129929: Create activitystream list view for clients

### CODE REVIEW

File: src/admin/models/activities.php [hunk 1]
```diff
@@ -113,7 +122,7 @@ protected function getListQuery()
 		}
 
 		// Add the list ordering clause.
-		$orderCol = $this->state->get('list.ordering', 'created_date');
+		$orderCol  = $this->state->get('list.ordering', 'created_date');
```
Comment (ankush-maherwal) [inline src/admin/models/activities.php]: @Mayaank12 

extra space before =

File: src/admin/models/activities.php [hunk 2]
```diff
@@ -130,9 +139,9 @@ protected function getListQuery()
 	 */
 	public function getItems()
 	{
-		$items = parent::getItems();
+		$items                          = parent::getItems();
```
Comment (ankush-maherwal) [inline src/admin/models/activities.php]: @Mayaank12 
remove extra lines

File: src/admin/models/fields/activitytype.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @version    SVN: <svn_id>
```
Comment (ankush-maherwal) [inline src/admin/models/fields/activitytype.php]: @Mayaank12 
update the comment

File: src/admin/models/fields/activitytype.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+// No direct access.
+defined('_JEXEC') or die();
+
+JFormHelper::loadFieldClass('list');
+
+/**
+ * Supports an HTML select list of courses
+ *
+ * @since  1.0.0
```
Comment (ankush-maherwal) [inline src/admin/models/fields/activitytype.php]: @Mayaank12 
Check comments

File: src/admin/models/fields/activitytype.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		$db     = JFactory::getDbo();
+		$input  = JFactory::getApplication()->input;
+		$client = $input->get('client', '', 'STRING');
+		$filter = JFilterInput::getInstance();
+
+		$query = $db->getQuery(true);
+		$query->select('distinct l.type');
+		$query->from('#__tj_activities AS l');
+		$query->where($db->quoteName('client') . ' = ' . $db->quote($client));
+		$query->order($db->escape('l.type ASC'));
```
Comment (ankush-maherwal) [inline src/admin/models/fields/activitytype.php]: @Mayaank12 
use query builder

File: src/admin/models/fields/activitytype.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+		$db->setQuery($query);
+
+		// Send filter values
+		$dataType  = $db->loadObjectList();
```
Comment (ankush-maherwal) [inline src/admin/models/fields/activitytype.php]: @Mayaank12 
$dataType => $activityType

File: src/admin/models/fields/activitytype.php [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+		$options   = array();
+		$options[] = JHtml::_('select.option', 'all', JText::_('COM_ACTIVITYSTREAM_SEARCH_FILTER'));
+
+		foreach ($dataType as $type)
+		{
+			$tempArr     = implode("_", explode('.', $type->type));
```
Comment (ankush-maherwal) [inline src/admin/models/fields/activitytype.php]: @Mayaank12 
check the variable name 

it should not be array as it stores string

File: src/admin/models/forms/filter_activities.xml [hunk 8]
```diff
@@ -16,6 +16,16 @@
 			onchange="this.form.submit();"
 			>
 			<option value="">JOPTION_SELECT_PUBLISHED</option>
+
```
Comment (ankush-maherwal) [inline src/admin/models/forms/filter_activities.xml]: @Mayaank12 
remove extra blank lines

File: src/admin/sql/install.mysql.utf8.sql [hunk 9]
```diff
@@ -7,6 +7,7 @@ CREATE TABLE IF NOT EXISTS `#__tj_activities` (
 	`target` text(900),
 	`target_id` text(900),
 	`type` text(900),
+	`client` text(900),
```
Comment (ankush-maherwal) [inline src/admin/sql/install.mysql.utf8.sql]: data type for client should be varchar and not text

File: src/admin/views/activities/tmpl/default.php [hunk 10]
```diff
@@ -15,7 +15,7 @@
 $listOrder     = $this->escape($this->state->get('list.ordering'));
 $listDirn      = $this->escape($this->state->get('list.direction'));
 ?>
-<form action="index.php?option=com_activitystream&view=activities" method="post" id="adminForm" name="adminForm">
+<form action="<?php echo JRoute::_('index.php?option=com_activitystream&view=activities&client=' . $this->client); ?>"  method="post" id="adminForm" name="adminForm">
```
Comment (ankush-maherwal) [inline src/admin/views/activities/tmpl/default.php]: @Mayaank12 
no need to route URL in backend

File: src/admin/views/activities/view.html.php [hunk 11]
```diff
@@ -60,7 +69,14 @@ public function display($tpl = null)
 	 */
 	protected function addToolBar()
 	{
-		$title = JText::_('Activity Stream');
+		$title        = JText::_('Activity Stream');
+		$input        = JFactory::getApplication()->input;
+		$languageFile = JFactory::getLanguage();
+		$extension    = JFactory::getApplication()->input->get('client', '', 'word');
```
Comment (ankush-maherwal) [inline src/admin/views/activities/view.html.php]: @Mayaank12 check the datatype (third param)

File: src/admin/views/activities/view.html.php [hunk 12]
```diff
@@ -74,6 +90,10 @@ protected function addToolBar()
 		JToolBarHelper::publish('activity.publish');
 		JToolBarHelper::unpublish('activity.publish');
 		JToolBarHelper::save2copy('activity.save2copy');
+
+		JHtmlSidebar::setAction('index.php?option=com_activitystream&view=activities');
+
+		$this->extra_sidebar = '';
```
Comment (ankush-maherwal) [inline src/admin/views/activities/view.html.php]: @Mayaank12 
Do we need this?

File: src/admin/models/activities.php [hunk 13]
```diff
@@ -54,12 +54,21 @@ public function __construct($config = array())
...
+		$this->setState('filter.component', $parts[0]);
+		$client = $parts[0];
+
+		$db     = JFactory::getDbo();
+		$query  = $db->getQuery(true);
 
 		// Create the base select statement.
 		$query->select($this->getState('list.select', '*'))
 			->from($db->quoteName('#__tj_activities'));
+			$query->where($db->quoteName('client') . ' = ' . $db->quote($extension));
```
[the remark this answers is not in the document]
Comment (Mayaank12) [reply src/admin/models/activities.php]: Done

File: src/admin/models/activities.php [hunk 14]
```diff
@@ -78,16 +87,16 @@ protected function getListQuery()
...
+		$from_date  = $this->getState('from_date');
+		$limit      = $this->getState('limit');
 
 		$result_arr = array();
 
 		// Return result related to specified activity type
 		if (!empty($type) && $type != 'all')
 		{
-			$query->where($db->quoteName('type') . ' IN (' . $type . ')');
+			$query->where($db->quoteName('type') . ' = ' . $db->quote($type));
```
[the remark this answers is not in the document]
Comment (Mayaank12) [reply src/admin/models/activities.php]: Done

File: src/admin/views/activities/view.html.php (hunk 12 above)
Comment (Mayaank12) [reply src/admin/views/activities/view.html.php]: no

File: src/admin/models/activities.php (hunk 2 above)
Comment (Mayaank12) [reply src/admin/models/activities.php]: Done

File: src/admin/models/fields/activitytype.php (hunk 5 above)
Comment (Mayaank12) [reply src/admin/models/fields/activitytype.php]: done

File: src/admin/models/fields/activitytype.php (hunk 7 above)
Comment (Mayaank12) [reply src/admin/models/fields/activitytype.php]: done

File: src/admin/views/activities/view.html.php (hunk 11 above)
Comment (Mayaank12) [reply src/admin/views/activities/view.html.php]: done

File: src/admin/models/forms/filter_activities.xml (hunk 8 above)
Comment (Mayaank12) [reply src/admin/models/forms/filter_activities.xml]: done

File: src/admin/models/fields/activitytype.php (hunk 6 above)
Comment (Mayaank12) [reply src/admin/models/fields/activitytype.php]: done

File: src/admin/views/activities/tmpl/default.php (hunk 10 above)
Comment (Mayaank12) [reply src/admin/views/activities/tmpl/default.php]: Done

File: src/admin/models/fields/activitytype.php (hunk 4 above)
Comment (Mayaank12) [reply src/admin/models/fields/activitytype.php]: Done

File: src/admin/sql/install.mysql.utf8.sql (hunk 9 above)
Comment (Mayaank12) [reply src/admin/sql/install.mysql.utf8.sql]: Done

File: src/admin/models/fields/activitytype.php (hunk 3 above)
Comment (Mayaank12) [reply src/admin/models/fields/activitytype.php]: Done

File: src/admin/controllers/activities.php [hunk 15]
```diff
@@ -1,18 +1,18 @@
...
 // No direct access to this file
 defined('_JEXEC') or die;
+use Joomla\Utilities\ArrayHelper;
 
 /**
- * HelloWorlds Controller
+ * Activity Stream Controller
  *
- * @since  0.0.1
+ * @since  1.0.2
```
Comment (ankush-maherwal) [inline src/admin/controllers/activities.php]: @Mayaank12 
since version should be 0.0.1

File: src/admin/controllers/activities.php [hunk 16]
```diff
@@ -94,13 +94,70 @@ public function getActivities()
...
+			// Make sure the item ids are integers
+			jimport('joomla.utilities.arrayhelper');
+			ArrayHelper::toInteger($id);
+
+			// Remove the activity.
+			if ($model->delete($id))
+			{
+				$this->setMessage(JText::plural($this->text_prefix . '_N_ITEMS_DELETED', count($id)));
+
+				if (isset($client))
```
Comment (ankush-maherwal) [inline src/admin/controllers/activities.php]: @Mayaank12 
duplicate code, can be moved out of if else and can be written once

File: src/admin/models/activities.php [hunk 17]
```diff
@@ -71,16 +119,30 @@ protected function getListQuery()
...
 
 		// Filter by published state
-		$published = $this->getState('filter.state');
+		$published = $this->getState('filter.published');
 
 		if (is_numeric($published))
 		{
 			$query->where('state = ' . (int) $published);
 		}
+		elseif ($published === '')
```
Comment (ankush-maherwal) [inline src/admin/models/activities.php]: @Mayaank12 
state can only be 0 OR 1

Do we need this elseif condition?

File: src/admin/models/fields/activitytype.php [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+
+// No direct access.
+defined('_JEXEC') or die();
+
+JFormHelper::loadFieldClass('list');
+
+/**
+ * Supports an HTML select list of courses
+ *
+ * @since  1.1.0
```
Comment (ankush-maherwal) [inline src/admin/models/fields/activitytype.php]: @Mayaank12 
check since version

File: src/admin/controllers/activities.php (hunk 15 above)
Comment (Mayaank12) [reply src/admin/controllers/activities.php]: Done

File: src/admin/models/activities.php (hunk 17 above)
Comment (Mayaank12) [reply src/admin/models/activities.php]: No

File: src/admin/controllers/activities.php (hunk 16 above)
Comment (Mayaank12) [reply src/admin/controllers/activities.php]: Done

File: src/admin/models/fields/activitytype.php (hunk 18 above)
Comment (Mayaank12) [reply src/admin/models/fields/activitytype.php]: Done

**Pull request merged**